google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/it/ExecuteQueryIT.java

```diff
@@ -0,0 +1,318 @@
+                  .get(ByteString.copyFromUtf8("qual")))
+          .isEqualTo(ByteString.copyFromUtf8("bval"));
+
+      assertThat(rs.next()).isFalse();
+    }
+  }
+
+  @Test
+  public void withHistoryQuery() {
+    try (ResultSet rs =
+        dataClient.executeQuery(
+            Statement.of(
+                "SELECT * FROM `"
+                    + tableId
+                    + "`(with_history => true) WHERE _key LIKE '"
+                    + uniquePrefix
+                    + "%'"))) {
+
+      assertThat(rs.next()).isTrue();
+      assertThat(rs.getBytes("_key")).isEqualTo(ByteString.copyFromUtf8(uniquePrefix + "a"));
+      Map<ByteString, List<Struct>> rowACf = rs.getMap(cf, SqlType.historicalMap());
+      List<Struct> rowAQual = rowACf.get(ByteString.copyFromUtf8("qual"));
+      assertThat(rowAQual.size()).isEqualTo(2);
+      Struct rowAQual_0 = rowAQual.get(0);
+      assertThat(rowAQual_0.getBytes("value")).isEqualTo(ByteString.copyFromUtf8("val"));
+      // timestamp in micros above so we divide by 1000
+      assertThat(rowAQual_0.getTimestamp("timestamp")).isEqualTo(Instant.ofEpochMilli(10));
+      Struct rowAQual_1 = rowAQual.get(1);
+      assertThat(rowAQual_1.getBytes("value")).isEqualTo(ByteString.copyFromUtf8("old"));
+      assertThat(rowAQual_1.getTimestamp("timestamp")).isEqualTo(Instant.ofEpochMilli(1));
+
+      assertThat(rs.next()).isTrue();
+      assertThat(rs.getBytes("_key")).isEqualTo(ByteString.copyFromUtf8(uniquePrefix + "b"));
+      Map<ByteString, List<Struct>> rowBCf = rs.getMap(cf, SqlType.historicalMap());
+      List<Struct> rowBQual = rowBCf.get(ByteString.copyFromUtf8("qual"));
+      assertThat(rowBQual.size()).isEqualTo(1);
+      Struct rowBQual_0 = rowBQual.get(0);
+      assertThat(rowBQual_0.getBytes("value")).isEqualTo(ByteString.copyFromUtf8("bval"));
+
+      assertThat(rs.next()).isFalse();
+    }
+  }
+
+  @Test
+  public void allTypes() {
+    try (ResultSet rs =
+        dataClient.executeQuery(
+            Statement.of(
+                "SELECT 'stringVal' AS strCol, b'foo' as bytesCol, 1 AS intCol, CAST(1.2 AS FLOAT32) as f32Col, "
+                    + "CAST(1.3 AS FLOAT64) as f64Col, true as boolCol, TIMESTAMP_FROM_UNIX_MILLIS(1000) AS tsCol, "
+                    + "DATE(2024, 06, 01) as dateCol, STRUCT(1 as a, \"foo\" as b) AS structCol, [1,2,3] AS arrCol, "
+                    + cf
+                    + " as mapCol FROM `"
+                    + tableId
+                    + "` WHERE _key='"
+                    + uniquePrefix
+                    + "a' LIMIT 1"))) {
+
+      assertThat(rs.next()).isTrue();
+      assertThat(rs.getString("strCol")).isEqualTo("stringVal");
+      assertThat(rs.getString(0)).isEqualTo("stringVal");
+      assertThat(rs.getBytes("bytesCol")).isEqualTo(ByteString.copyFromUtf8("foo"));
+      assertThat(rs.getBytes(1)).isEqualTo(ByteString.copyFromUtf8("foo"));
+      assertThat(rs.getLong("intCol")).isEqualTo(1L);
```

Comment:
Lets add a column that is only present in one row and validate the `rs.getStirng(non_existant_col)` works as expected.

Reply:
Done. Added in a separate test below for simplicity